src/esp/gfx_batch/DepthUnprojection.cpp

```diff
@@ -127,5 +126,5 @@ void unprojectDepth(const Mn::Vector2& unprojection,
   }
 }
```

Comment:
Continuing from my [earlier comment](https://github.com/facebookresearch/habitat-sim/pull/2129#discussion_r1230764089):

```cpp
void unprojectDepth(const Mn::Vector2& unprojection,
                    const Cr::Containers::StridedArrayView2D<Mn::Float>& depth) {
  for (Cr::Containers::StridedArrayView1D<Mn::Float> row : depth) {
    for (Mn::Float& d : row) {
      d = unprojection[1] / (d + unprojection[0]);
    }
  }
  /* Change pixels on the far plane to be 0. Done in a separate loop to allow
     the optimizer to vectorize the above better.  */
  const Mn::Float farDepth = unprojection[1] / (1.0f + unprojection[0]);
  for (Cr::Containers::StridedArrayView1D<Mn::Float> row : depth) {
    for (Mn::Float& d : row) {
      /* We can afford using == for comparison as 1.0f has an exact
         representation, the depth was cleared to exactly this value and the
         calculation is done exactly the same way in both cases -- thus the
         result should be bit-exact. */
      if (d == farDepth)
        d = 0.0f;
    }
  }
}
```

Then you'd call this function not on `Cr::Containers::arrayCast<Mn::Float>(image.data())`, but rather on `image.pixels<Mn::Float>()`. That's all that is needed to change (plus fixing the test to slice in two dimensions).